### ISSUE TITLE
Permission issue with saving pages

### ISSUE THREAD

**Issue by kriskropd** [yomli/pico-plugins#5]
I've been able to get image assets to upload (I did have to manually create the `images` directory and the contributors subdirectory as well as set the permissions, but it is uploading image assets with no apparent errors) however I'm still having trouble debugging why pages will not save.

> "POST /contribute/save HTTP/1.1" 200 52 "http://site.com/contribute" "Mozilla/5.0 (X11; Linux x86_64; rv:40.0) Gecko/20100101 Firefox/40.0"

When pressing the Save button, that is the only new line that appears in the logfile while the red "Saving..." message remains indefinitely on the browser page. I've applied the same permissions to the `content` directory as the `content/images/contributor` directory (which, as I mentioned, appears to be working for image uploads).

**Comment by yomli** [comment]
Not a permission issue, I guess. Your "Saving…" message remains. Two ways to do that :
1. `file_put_contents` is looping so PHP can't respond anything to JS… Very unlikely since, even in case of permission issue, `file_puts_contents` returns `FALSE` (and you would notice something in the logfile).
2. The content of `$_POST['content']` sent by JS is null, or `$_POST['content']` is not set. In which case, PHP dies and JS alerts you with "Error: Invalid content".

So I think it's the second one, but why the browser doesn't alert ?

**Comment by kriskropd** [comment]
Okay, I've uncovered why the "Saving" message remained - I had written my own metadata thinking that would suffice instead of using the "Create a new post" button from the left Menu. Monitoring the network tab in the Firefox Developer Tools showed that for the POST `save` request to http://site.com/contribute/save were receiving a `Error: Invalid file:` response. I didn't see it before because I thought it might be server-side, permission related. If I use the Menu>Create a new post button to send a POST `new` request first, that "Saving message" will dissapear shortly after hitting the Save button. As a suggestion, maybe give the option for someone to make a `new` page/post when saving text in the editor that loads blank at default. Just an idea for user experience.

While saving pages that have been **just** generated seemed to be working, I've found a new problem for retrieving said pages from the Menu. New post/page .md files are made in the expected 'content' directory on the server, but when trying to _retrieve_ a post/page from the list on the Menu the Editor instead prints

> Error: Invalid file: /usr/share/nginx/site.com/content/site.com/20150826102414-user.md

Notice the path is repeating 'site.com' twice. However, when creating the new post/page above it had saved to /usr/share/nginx/site.com/content/20150826102414-user.md.

Not only are saved pages un-retrievable, but also when reloading the contributors page the menu doesn't locate any previous pages to open either. So, retrieving pages must be including the base_url twice, once before and once after the 'content' directory, but this doesn't occur when saving files.
